scene/gui/line_edit.cpp

```diff
@@ -724,15 +724,14 @@ void LineEdit::_notification(int p_what) {
 			int font_ascent = font->get_ascent();
 
 			Color selection_color = get_color("selection_color");
-			Color font_color = get_color("font_color");
+			Color font_color = is_editable() ? get_color("font_color") : get_color("font_color_uneditable");
 			Color font_color_selected = get_color("font_color_selected");
 			Color cursor_color = get_color("cursor_color");
 
 			const String &t = using_placeholder ? placeholder : text;
 			// draw placeholder color
 			if (using_placeholder)
 				font_color.a *= placeholder_alpha;
-			font_color.a *= disabled_alpha;
```

Comment:
I think you can remove the `disabled_alpha` variable which is now only used once at line 740 (you can use a ternary operator there instead to set the alpha).

Reply:
I felt a bit weird about the ternary bit, does it look okay?

Reply:
I would have simplified to `is_editable() ? 0.9 : 0.45`, but it's also ok as is.